mautic/mautic
# Email tracking links are being incorrectly assigned to inappropriate contacts

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Yes
| New feature/enhancement? (use the a.x branch)      | No
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | No 
| Related user documentation PR URL      |  
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 



#### Description:
Same email tracking links was being assigned to all contacts if the unsubscribe link was invalid like `<a id="{unsubscribe_url}">`. Notice the `id` where `href` should be.

**Root cause :** 
Event subscribe `Mautic\EmailBundle\EventListener\BuilderSubscriber::fixEmailAccessibility()` has added subject into email title so in this case email content has been changed and it changed the [contentHash](https://github.com/mautic/mautic/blob/5.x/app/bundles/EmailBundle/Helper/MailHelper.php#L1005) and due to this it does not use [existing manipulated html content](https://github.com/mautic/mautic/blob/5.x/app/bundles/EmailBundle/EventListener/BuilderSubscriber.php#L410) and it used old trackable url for all further contacts.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new custom field called "Url"
3. Add some valid URL to at least 2 contacts to this new field. The URL must be unique for each contact.
4. Create a segment with filter on the Url field not empty
5. Create a new segment email with this content:
`<!DOCTYPE html><htm><body><a href="{contactfield=url}">link</a><a id="{unsubscribe_url}">unsubscribe here</a></body></html>`
6. Send the email
7. Open the email link for both contacts in an anonymous window and verify that both links should not lead to the same URL even though the contacts have different URLs in the field.

#### Other areas of Mautic that may be affected by the change:
1. Just email URLs

#### List deprecations along with the new alternative:
1. None



[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. Functional test that is testing to send email with a unsubscribe link without href
